### PR TITLE
add namespace check for uid/gid mappings

### PR DESCRIPTION
**Pull request by Mashimiao**
Signed-off-by: Ma Shimiao <mashimiao.fnst@cn.fujitsu.com>

### CODE REVIEW

File: cmd/ocitools/validate.go [hunk 1]
```diff
@@ -340,6 +336,14 @@ func checkLinux(spec rspec.Spec, rootfs string, hostCheck bool) (msgs []string)
 		}
 	}
 
+	if (len(spec.Linux.UIDMappings) > 0 || len(spec.Linux.GIDMappings) > 0) && !userExists {
+		msgs = append(msgs, "UID/GID mappings require User namespace exists")
```
Comment (wking) [inline cmd/ocitools/validate.go]: The phrasing used by existing similar warnings is:

> {setting} requires a new {type} namespace to be specified as well

Comment (Mashimiao) [inline cmd/ocitools/validate.go]: On 08/24/2016 12:50 AM, W. Trevor King wrote:
> The phrasing used by existing similar warnings is:
> 
>     {setting} requires a new {type} namespace to be specified as well
> 
I think container joins into an existing user namespace also can work.
Doesn't it?

Comment (wking) [inline cmd/ocitools/validate.go]: On Tue, Aug 23, 2016 at 05:58:31PM -0700, Ma Shimiao wrote:
> > +	if (len(spec.Linux.UIDMappings) > 0 || len(spec.Linux.GIDMappings) > 0) && !userExists {
> > +		msgs = append(msgs, "UID/GID mappings require User namespace exists")
> 
> 08/24/2016 12:50 AM, W. Trevor King:
> > The phrasing used by existing similar warnings is:
> > 
> >     {setting} requires a new {type} namespace to be specified as well
>
> I think container joins into an existing user namespace also can
> work.  Doesn't it?

The kernel has no problem with join-and-tweak, but the OCI spec does
not allow it at the moment [1,2].  Interestingly, the spec places no
such restriction on namespaces inherited from the host.  I've filed
[3] to figure out what the runtime-spec maintainers want to do about
that.

[1]: https://github.com/opencontainers/runtime-spec/blame/v1.0.0-rc1/config-linux.md#L41
[2]: https://github.com/opencontainers/runtime-spec/pull/158
[3]: https://github.com/opencontainers/runtime-spec/issues/537

Comment (Mashimiao) [inline cmd/ocitools/validate.go]: Thanks, got it. 
Since spec does not allow now, I fixed error message as you suggested.

**Comment by wking** [comment]
bf50b73b looks good to me.  It's taking the (b) approach from [1],
even though the current spec wording doesn't quite match that
interpretation.  But it matches the ocitools existing namespace
validation logic, and we can revisit all of that if [1] ends up being
resolved in a different direction.

[1]: https://github.com/opencontainers/runtime-spec/issues/537

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**